code/modules/library/lib_items.dm

```diff
@@ -96,6 +96,56 @@
 		icon_state = "book-5"
 
 
+/obj/structure/bookcase/libraryspawn
+	var/spawn_category
+	var/spawn_amount = 3
+
+/obj/structure/bookcase/libraryspawn/Initialize()
+
+	. = ..()
+
+	name = "[initial(name)] ([spawn_category])"
+
+	if(!config.sql_enabled)
+		return
+
+	establish_db_connection(dbcon)
+
+	if(!dbcon.IsConnected())
+		log_debug("Library Book Spawning: SQL ERROR - Failed to connect and spawn books for [spawn_category].")
+		return
+
+	var/cat_fix = spawn_category ? "category = :cat:" : ""
+	var/DBQuery/query_books = dbcon.NewQuery("SELECT * FROM ss13_library WHERE category = [cat_fix] ORDER BY RAND() LIMIT :amount:")

```

Comment:
This is like, five levels of haram.

```dm
/obj/structure/bookcase/libraryspawn/Initialize()
	. = ..()
	name = "[initial(name)] ([spawn_category])"

	if (!establish_db_connection(dbcon))
		return

	var/query_str = "SELECT * FROM ss13_library ORDER BY RAND() LIMIT :amount:"
	var/list/query_data = list("amount" = amount)

	if (spawn_category)
		query_str = "SELECT * FROM ss13_library WHERE category = :cat: ORDER BY RAND() LIMIT :amount:"
		query_data["cat"] = spawn_category

	var/DBQuery/query_books = dbcon.NewQuery(query_str)
	query_books.Execute()

	while (query_books.NextRow())
		// blablabla

	update_icon()
```

Reply:
A guide on how to SQL in DM/SS13 can be found here: https://github.com/Aurorastation/Aurora.3/wiki/MySQL